server.js

```diff
@@ -1540,13 +1540,24 @@ cache(function(data, match, sendBadge, request) {
   var userRepo = match[1];  // eg, `github/codecov/example-python`.
   var branch = match[2];
   var format = match[3];
+  var token = data['token'];  // for private repositories (?token=1234)
```

Comment:
As discussed in the issue, I'd like to reserve `data` for switches that work on all badges, including custom badges (ie, `/badge/my-data-green.svg`). Otherwise, it's a nightmare for me to determine whether a global switch I want to add doesn't clash with something used.

Additionally, `data` information is necessary for caching, so here this badge would be cached as not having a token, and would potentially fail, or make another badge fail.